controllers/reconciler.go

```diff
@@ -132,6 +132,20 @@ func (r *SingleClusterReconciler) Reconcile() (ctrl.Result, error) {
 		return reconcile.Result{}, err
 	}
 
+	// Use policy from spec after setting up access control
+	policy := r.getClientPolicyFromSpec()
+
+	// TODO: Should we just Wait for migration before setting roster only in scale down?
+	if res := r.waitForClusterStability(policy, allHostConns); !res.isSuccess {
```

Comment:
Makes sense to set this only in case of scale down. Is there any issue in setting roster while migration is going on?

Reply:
1: Yeah, waiting for it only after scaling down seems logical. But as per the Server team's suggestion, it's good practice to always wait for migration before setting the roster.

Reply:
We can remove the TODO then